src/Basket.sol

```diff
@@ -612,17 +625,58 @@ contract Basket is Initializable, RebaseTokenUpgradeable, IBasket, IRWAPriceNoti
         sharePrice = (getTotalValueOfBasket() * 10 ** decimals()) / totalSupply();
     }
 
+    /**
+     * @dev Get $USD Value of specified token.
+     * @param _tangibleNFT TNFT contract address.
+     * @param _tokenId TokenId of token.
+     * @return $USD value of token, note: base 1e18
+     */
+    function getUSDValue(address _tangibleNFT, uint256 _tokenId) public view returns (uint256) {
+        return IBasketManager(basketManager).currencyCalculator().getUSDValue(_tangibleNFT, _tokenId);
+    }
+
     /**
      * @notice This method returns the total value of NFTs and claimable rent in this basket contract.
      * @return totalValue -> total value in 18 decimals.
      */
     function getTotalValueOfBasket() public view returns (uint256 totalValue) {
-        // get total value of nfts in basket.
-        totalValue = totalNftValue;
+        // get total value of nfts in basket by currency
+        uint256 len = supportedCurrencies.length;
+        for (uint256 i; i < len; ++i) {
+            string memory currency = supportedCurrencies[i];
+            if (totalNftValueByCurrency[currency] != 0) {
```

Comment:
extract totalNftValueByCurrency[currency] to storage variable

Reply:
Resolved